crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -121,7 +122,11 @@ declare_lint! {
 }
 
 declare_lint! {
-    /// TODO #14889
+    /// ## What it does
+    /// Checks whether a variable has been declared as two conflicting types.
+    ///
+    /// ## Why is this bad
+    /// A variable with two conflicting declarations likely indicates a mistake. Moreover, it could lead to incorrect or ill-defined type inference for other code that relies on these variables.
```

Comment:
nit: we usually wrap these at 80 or 100 columns (seems like there's some of both in the file, probably depends who wrote them :) )